src/job_cache/daemon_cache.cpp

```diff
@@ -47,17 +47,66 @@ namespace {
 using namespace job_cache;
 
 static std::unique_ptr<std::ofstream> initialize_logging() {
+  const char *str_fmt = "%Y-%m-%d";
+  const size_t str_fmt_len = 10;         // count("XXXX-XX-XX")
+  const size_t filename_prefix_len = 7;  // count (".cache.")
+  const size_t filename_suffix_len = 4;  // count (".log")
+
   time_t today = time(NULL);
   struct tm tm = *localtime(&today);
-  char time_buffer[10 + 1];
-  strftime(time_buffer, sizeof(time_buffer), "%Y-%m-%d", &tm);
+  char time_buffer[str_fmt_len + 1];
+  strftime(time_buffer, sizeof(time_buffer), str_fmt, &tm);
   std::string log_path = ".cache." + std::string(time_buffer) + ".log";
 
   std::unique_ptr<std::ofstream> log_file =
       std::make_unique<std::ofstream>(log_path, std::ios::app);
   wcl::log::subscribe(std::make_unique<wcl::log::FormatSubscriber>(log_file->rdbuf()));
 
   wcl::log::info("Initialized logging for job cache daemon")();
+
+  auto res = wcl::directory_range::open(".");
+  if (!res) {
+    wcl::log::warning("Failed to open cwd to cleanup log files")();
+    return log_file;
+  }
+
+  std::vector<std::string> to_delete;
+  for (const auto &entry : *res) {
+    if (!entry) {
+      wcl::log::warning("bad /proc/self/fd/ entry: %s\n", strerror(entry.error()))();
+      continue;
+    }
+
+    // Only consider files
+    if (entry->type != wcl::file_type::regular) continue;
+    // Don't consider file names that aren't the correct size
+    // count(".cache.XXXX-XX-XX.log") == 21
+    if (entry->name.size() != filename_prefix_len + str_fmt_len + filename_suffix_len) continue;
+    // Only consider files that start with ".cache."
+    if (entry->name.find(".cache.") != 0) continue;
+    // and end in ".log"
+    if (entry->name.find(".log") != filename_prefix_len + str_fmt_len) continue;
+    // Don't consider the current log file
+    if (entry->name == log_path) continue;
+
+    std::string day = entry->name.substr(filename_prefix_len, str_fmt_len);
+    struct tm prev_tm = {0};
+    strptime(day.c_str(), str_fmt, &prev_tm);
+
+    double diff_secs = difftime(today, mktime(&prev_tm));
+
+    int four_days = 60 /*secs*/ * 60 /*mins*/ * 24 /*hours*/ * 4 /*days*/;
+    if (diff_secs > four_days) {
+      to_delete.push_back(entry->name);
```

Comment:
nit: if you have time to get to this modtime would be better because 1) it would actually be 3-days since it was touched and 2) it would be more robust than parsing from the file name